### PR TITLE
Weighted target groups upon ALB.

### PR DESCRIPTION
**Pull request by IYermakov**
## What Are We Doing Here?

PR introduces support of weighted target groups on ALB.

## How to Test and Verify

1. Check out this PR
2. Follow the instructions in README file

### Level

`Medium`

### Required Testing

`Smoke`, `Manual`

### Risk Summary

By default, it assumes an application has to be deployed behind the "Blue" target group. If no, you have to change weights, accordingly to your expectations.

## Code Review Checklist

- [ ] Testing instructions are included. If not, make sure you leave us a line or two for the reason.

- [ ] Well documented, including updates to any necessary markdown files. When
we inevitably come back to this code it will only take hours to figure out, not
days.

- [ ] Make sure PR has only the changes required for the Jira issues.

- [ ] Have you make sure to follow the agreements in our process document?

- [ ] Have someone from different region than you review your changes. If not, make sure to leave a comment on why it is not required

- [ ] If it's applicable, document deployment plan for the changes.

### CODE REVIEW

File: ops/cloudformation/alb.yml [hunk 1]
```diff
@@ -1,24 +1,56 @@
...
+# aws cloudformation deploy --stack-name alb \
+#                           --template-file ops/cloudformation/alb.yml \
+#                           --parameter-overrides VPCStackName=${VPCStackName} \
+#                                                 Environment=${Environment} \
+#                                                 SSLCertificateARN=${SSLCertificateARN}
+---
+AWSTemplateFormatVersion: "2010-09-09"
 
-  Description: "Creates: ALB Security Group, App Load Balacer,
+Description: "Creates: ALB Security Group, App Load Balacer,
```
Comment (mike-unitskyi) [inline ops/cloudformation/alb.yml]: typo `Balacer`

File: ops/cloudformation/alb.yml [hunk 2]
```diff
@@ -1,24 +1,56 @@
...
-      AllowedValues: [ Dev, Prod ]
-    SSLCertificateARN:
-      Type: String
-      Description: ARN of existing SSL Certificate.
-
-  Resources:
-    ALBSecurityGroup:
+Parameters:
+
+  VPCStackName:
```
Comment (mike-unitskyi) [inline ops/cloudformation/alb.yml]: does it make sense to have map with `VPCStackNames` and chose based on `Environment` parameter?

File: ops/cloudformation/alb.yml [hunk 3]
```diff
@@ -1,24 +1,56 @@
...
+    AllowedValues: [ Dev, Prod ]
+
+  SSLCertificateARN:
+    Type: String
+    Description: ARN of existing SSL Certificate.
+
+  BlueWeight:
+    Type: String
+    Default: 1
+    Description: Green target group weight. 100% by default. This parameter is on canary deployment abidance.
```
Comment (IYermakov) [inline ops/cloudformation/alb.yml]: Need to fix

File: ops/cloudformation/alb.yml (hunk 1 above)
Comment (IYermakov) [reply ops/cloudformation/alb.yml]: fixed

File: ops/cloudformation/alb.yml (hunk 2 above)
Comment (IYermakov) [reply ops/cloudformation/alb.yml]: At the moment no. If we decided to introduce one more environment, we'd do it.

File: ops/cloudformation/alb.yml (hunk 3 above)
Comment (IYermakov) [reply ops/cloudformation/alb.yml]: fixed

File: ops/cloudformation/alb.yml (hunk 2 above)
Comment (mike-unitskyi) [reply ops/cloudformation/alb.yml]: in this case we have code duplication. would like to see at least github issue if comment won't be addressed in this PR.

Comment (IYermakov) [reply ops/cloudformation/alb.yml]: Actually no, cause we were not going to restrict stack names. I've created an issue - https://github.com/IYermakov/DevOpsA3Training/issues/29

**Pull request merged**